apps/src/export.js

```diff
@@ -0,0 +1,138 @@
+  for (const fileName in files) {
+    if (fileName.indexOf(assetPrefix) !== 0) {
+      continue;
+    }
+    const relativePath = fileName.substring(assetPrefixLength);
+    moduleList.push({fileName: relativePath});
+  }
+  const entries = moduleList.map(module =>
+    exportExpoPackagedFilesEntryEjs({module})
+  );
+  return exportExpoPackagedFilesEjs({entries});
+}
+
+export async function generateExpoApk(options, config) {
+  const {appName, expoSnackId, iconUri, splashImageUri} = options;
+  const session = new SnackSession({
+    sessionId: `${getEnvironmentPrefix()}-${project.getCurrentId()}`,
+    name: `project-${project.getCurrentId()}`,
+    sdkVersion: '31.0.0',
+    snackId: expoSnackId,
+    user: {
+      sessionSecret: config.expoSession || EXPO_SESSION_SECRET
+    }
+  });
+
+  const appJson = JSON.parse(
+    exportExpoAppJsonEjs({
+      appName,
+      projectId: project.getCurrentId(),
+      iconPath: iconUri,
+      splashImagePath: splashImageUri
+    })
+  );
+
+  // TODO: remove the onlineOnlyExpo patching once getApkUrlAsync()
+  // properly supports our full app.json
+  const {
+    updates, // eslint-disable-line no-unused-vars
+    assetBundlePatterns, // eslint-disable-line no-unused-vars
+    packagerOpts, // eslint-disable-line no-unused-vars
+    ...onlineOnlyExpo
+  } = appJson.expo;
+  appJson.expo = onlineOnlyExpo;
+
+  const artifactUrl = await session.getApkUrlAsync(appJson);
+
+  return artifactUrl;
+}
+
+const soundRegex = /(\bsound:\/\/[-A-Z0-9+&@#\/%?=~_|!:,.;]*[-A-Z0-9+&@#\/%=~_|])/gi;
+
+export function extractSoundAssets(options) {
+  const {sources, rootRelativeAssetPrefix, zipAssetPrefix} = options;
+  const allSounds = [];
+  sources.forEach(source =>
+    allSounds.push(...(source.match(soundRegex) || []))
+  );
+  const uniqueSounds = [...new Set(allSounds)];
+  return uniqueSounds.map(soundProtocolUrl => {
+    const soundOriginUrl = assetPrefix.fixPath(soundProtocolUrl);
+    const filename = soundProtocolUrl.replace(assetPrefix.SOUND_PREFIX, '');
+    return {
+      url: soundOriginUrl,
+      rootRelativePath: rootRelativeAssetPrefix + filename,
+      zipPath: zipAssetPrefix + filename,
+      dataType: 'binary',
+      filename,
+      searchUrl: soundProtocolUrl
+    };
+  });
+}
+
+// TODO: for expoMode, replace spaces in asset filenames or wait for this fix
+// to make it into Metro Bundler:
+// https://github.com/facebook/react-native/pull/10365
+export function rewriteAssetUrls(appAssets, data) {
+  return appAssets.reduce(function(data, assetToDownload) {
+    const searchUrl = assetToDownload.searchUrl || assetToDownload.filename;
+    data = data.replace(
+      new RegExp(`["|']${assetToDownload.url}["|']`, 'g'),
+      `"${assetToDownload.rootRelativePath}"`
+    );
+    return data.replace(
+      new RegExp(`["|']${searchUrl}["|']`, 'g'),
+      `"${assetToDownload.rootRelativePath}"`
+    );
+  }, data);
+}
+
+export function getEnvironmentPrefix() {
```

Comment:
Not introduced in this PR, but wanted to call out that this big switch statement feels like something that should be a locals/globals.yml configuration setting and passed down to the client, rather than hard-coded in the client code here.